runtime/autoload/netrw.vim

```diff
@@ -5294,6 +5294,104 @@ fun! s:NetrwBrowseUpDir(islocal)
 "  call Dret("s:NetrwBrowseUpDir")
 endfun
 
+func s:redir()
+  " set up redirection (avoids browser messages)
+  " by default if not set, g:netrw_suppress_gx_mesg is true
+  if get(g:, 'netrw_suppress_gx_mesg', 1)
+    if &srr =~# "%s"
+      return printf(&srr, has("win32") ? "nul" : "/dev/null")
+    else
+      return &srr .. (has("win32") ? "nul" : "/dev/null")
+    endif
+  endif
+  return ''
+endfunc
+
+fun! netrw#Launch(args)
+   let args = trim(a:args)
+   if has('unix')
```

Comment:
It's unnecessary to check for the OS on every invocation, but too much of a hassle to tear the checks apart out of the function

Reply:
> Platform tests can be pushed all the way up out of function
bodies and a separate `netrw#Launch()` implementation written
for each supported case.

Yes, is it worth it?

> Unfortunately, a mix of indentation styles has accreted over
the years. At one time (v7.0131), the style used to be sw=1
and an extra leading space character for function bodies and
line continuation, and using no tabs for indentation.)

It's a mess. The file should also be split